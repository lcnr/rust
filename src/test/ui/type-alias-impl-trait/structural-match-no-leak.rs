#![feature(type_alias_impl_trait)]

type Bar = impl Send;

// While i32 is structural-match, we do not want to leak this information.
// (See https://github.com/rust-lang/rust/issues/72156)
const fn leak_free() -> Bar {
    7i32
}
const LEAK_FREE: Bar = leak_free();

fn leak_free_test() {
    match LEAK_FREE {
        LEAK_FREE => (),
        //~^ ERROR constant pattern depends on a generic parameter
        //~| ERROR constant pattern depends on a generic parameter
        _ => (),
    }
}

fn main() {}
