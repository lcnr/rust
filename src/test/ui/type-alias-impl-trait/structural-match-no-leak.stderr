error: constant pattern depends on a generic parameter
  --> $DIR/structural-match-no-leak.rs:14:9
   |
LL |         LEAK_FREE => (),
   |         ^^^^^^^^^

error: constant pattern depends on a generic parameter
  --> $DIR/structural-match-no-leak.rs:14:9
   |
LL |         LEAK_FREE => (),
   |         ^^^^^^^^^

error: aborting due to 2 previous errors

